web3/providers/ipc.py

```diff
@@ -91,6 +93,14 @@ def get_default_ipc_path(testnet=False):
         if os.path.exists(ipc_path):
             return ipc_path
 
+        base_trinity_path = Path('~').expanduser() / 'Library' / 'Application Support' / 'trinity'
```

Comment:
Is this really where `XDG` directories end up for osx users?

Reply:
Not sure, I was cribbing off of Parity since I don't have a Mac to test with.

According to [this stackoverflow post](https://stackoverflow.com/a/5084892/8412986) it might actually be `~/Library/trinity` instead.

We might want to use a more unique namespace on the trinity side, actually, like `~/Library/org.ethereum.trinity`. For Mac, Linux, everything.

Reply:
@carver This is the path on my machine(mac):
`/Users/voith/.local/share/trinity/mainnet/logs/trinity.log`

Reply:
@carver Below is the path on my mac machine:
`/Users/voith/.local/share/trinity/mainnet/jsonrpc.ipc`

Reply:
However for parity this is my path: 
`/Users/voith/Library/Application Support/io.parity.ethereum/jsonrpc.ipc`

Reply:
Weird, I think I might call that a trinity bug.

Reply:
@carver I'm ok considering the move, but only in the face of an objective reason to do so.  I really like the simplicity of XDG and it seems like an easier to find location for most developers.

Reply:
> but only in the face of an objective reason to do so

Fair enough, and I don't think I can speak for our Mac users preferences here, so I'll drop it for now. I'll make the change in this PR to look in the `~/.local` location on Mac.